### PR TITLE
[SCIP] bump version to 8.0.4

### PR DESCRIPTION
**Pull request by hedtke**
Specify library name and version:  **scip/8.0.4**

SCIP and SoPlex are coupled together: there is a 1-to-1 correspondence between the SCIP version and the SoPlex version, thus the new map in this recipe.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/scip/all/conanfile.py [hunk 1]
```diff
@@ -76,11 +80,10 @@ def config_options(self):
 
     def requirements(self):
         if self.options.with_gmp:
-            self.requires("gmp/6.2.1")
+            self.requires("gmp/6.3.0")
         if self.options.with_sym == "bliss":
             self.requires("bliss/0.77")
-        self.requires("soplex/6.0.3")
-        self.requires("zlib/1.2.13")
+        self.requires(f"soplex/{self.soplex_version_belonging_to_me[self.version]}")
```
Comment (RubenRBS) [inline recipes/scip/all/conanfile.py:86]: If this is 1 to 1, couldn't we use `self.version` as the version? Or are you opening the door to possibly using newer patches?

Comment (hedtke) [reply recipes/scip/all/conanfile.py:86]: It is 1-to-1, but Soplex 6.0.4 belongs to SCIP 8.0.4, so versions are not exactly the same.

Comment (RubenRBS) [reply recipes/scip/all/conanfile.py:86]: That makes sense, thanks for the extra insight :)

**Review by RubenRBS — APPROVED** [review]

File: recipes/scip/all/conanfile.py [hunk 2]
```diff
@@ -76,11 +80,10 @@ def config_options(self):
 
     def requirements(self):
         if self.options.with_gmp:
-            self.requires("gmp/6.2.1")
+            self.requires("gmp/6.3.0")
         if self.options.with_sym == "bliss":
             self.requires("bliss/0.77")
-        self.requires("soplex/6.0.3")
-        self.requires("zlib/1.2.13")
```
Comment (uilianries) [inline recipes/scip/all/conanfile.py:83]: @hedtke Why did you remove zlib? I see it listed in scip files: https://github.com/scipopt/scip/blob/v804/src/xml/xmldef.h#L52

Comment (hedtke) [reply recipes/scip/all/conanfile.py:83]: it is automatically included by soplex

Comment (uilianries) [reply recipes/scip/all/conanfile.py:83]: @hedtke please, revert that change. Each package should manage their own dependencies, getting a hide over other dependencies to include a package will create a problem related to the package ID.

Comment (hedtke) [reply recipes/scip/all/conanfile.py:83]: Re-added and also using ranges instead of a specific version

**Review by uilianries — APPROVED** [review]
LGTM

**Review by RubenRBS — APPROVED** [review]

**Comment by RubenRBS** [comment]
Thanks for your patience @hedtke :)
